src/components/assessment/index.tsx

```diff
@@ -320,9 +320,15 @@ class Assessment extends React.Component<IAssessmentProps, State> {
     }
     return (
       <NavLink
-        to={`/academy/${assessmentCategoryLink(
-          overview.category
-        )}/${overview.id.toString()}/${DEFAULT_QUESTION_ID}`}
+        to={{
+          pathname: `/academy/${assessmentCategoryLink(
+            overview.category
+          )}/${overview.id.toString()}/${DEFAULT_QUESTION_ID}`,
+          search:
+            overview.private && overview.status === AssessmentStatuses.not_attempted
+              ? `isPasswordRequired=true`
+              : ''
```

Comment:
Using url query params is a really bad way of checking if an assessment requires a password.

I still think we should just send a null password, get the backend to check if the assessment requires a password, and return an error code if there is a need.